packages/block-editor/src/components/rich-text/event-listeners/paste-handler.js

```diff
@@ -26,15 +26,10 @@ export default ( props ) => ( element ) => {
 			preserveWhiteSpace,
 			pastePlainText,
 		} = props.current;
-		const { ownerDocument } = element;
-		const { defaultView } = ownerDocument;
-		const { anchorNode, focusNode } = defaultView.getSelection();
-		const containsSelection =
-			element.contains( anchorNode ) && element.contains( focusNode );
 
 		// The event listener is attached to the window, so we need to check if
 		// the target is the element.
-		if ( ! containsSelection ) {
+		if ( event.target !== element ) {
```

Comment:
I think we just need to keep the code here as is. Otherwise, the following two problems will occur:

- Formatting is not applied when formatted text is copied continuously
- Links cannot be pasted into formatted text

This problem was reported in #63434, and I tried to solve this problem in #63779, but #63671 ended up solving this problem as result. Please see [this discussion](https://github.com/WordPress/gutenberg/pull/63779#discussion_r1685752350) for why such code is needed.

### trunk (expected)

https://github.com/user-attachments/assets/e5a61338-b2ef-4307-87c4-47de495af433

### This PR (not expected)

https://github.com/user-attachments/assets/168f9de0-7e7a-4a20-b156-b6b04580b9f3